### ISSUE TITLE
Error when fitting a model to a measurement file

### ISSUE THREAD

**Issue by ahaselsteiner** [ahaselsteiner/virocon#129]
Bug when performing a fit:
When one first selects a distribution and then selects another distribution one gets an error in case the first distribution had selected a depdency function for a parameter which the second distribution does not have.

Step 1:
![image](https://user-images.githubusercontent.com/11452055/39141410-0b3aedfa-4728-11e8-9dc9-7fc3886a5d20.png)

Step 2: (Change from Log-normal to Weibull which has only two of the three parameters)
![image](https://user-images.githubusercontent.com/11452055/39141450-2ac5ed82-4728-11e8-8a1a-10e5d4f357cf.png)

Resulting error:
![image](https://user-images.githubusercontent.com/11452055/39141521-56c8b7d4-4728-11e8-95f3-bd819fb8ec62.png)
![image](https://user-images.githubusercontent.com/11452055/39141538-6193b56a-4728-11e8-9aff-7ddce038431b.png)

Solution: 
When the distribution is changed the dependency function needs to be reset.

**Comment by topape** [comment]
This issue was solved at the 23th of April in issue#41. But at the moment it is not merged into the master branch.

**Comment by ahaselsteiner** [comment]
I don't think it was solved in issue#41. I just tried with the feature-branch and still got the error.
